csv-validation-tests/src/test/scala/steps/StepDefinitions.scala

```diff
@@ -1,61 +1,88 @@
+          Response.ok(csv)
+        case r if r.uri.path.endsWith(notFountEndsWith) =>
+          Response("Not found", StatusCode.NotFound)
+        case r if r.uri.path.endsWith(notFoundStartsWith) =>
+        Response("Not found", StatusCode.NotFound)
+      })
+  }
 
   Given("""^I have a CSV file called "(.*?)"$""") { (filename: String) =>
-    //    log.info("Log me here..") Not working, need to fix
     val filePath = fixturesPath + filename
     val csv = Source.fromFile(filePath).getLines.mkString
   }
 
   Given("""^it is stored at the url "(.*?)"$""") { (url: String) =>
-    // Implemented using stub_request in ruby implementation
-    // Need to figure out purpose of this stub request, may be called during validation later?
+    contextUrl = url
+    // notFoundEndsWith is a list which holds all the url endwith strings for which a 404 should be returned.
+    // This list is later used in setTestingBackend function to mock http calls
+    notFountEndsWith = "/.well-known/csvm" :: notFountEndsWith
+    notFountEndsWith = "-metadata.json" :: notFountEndsWith
+    notFountEndsWith = "csv-metadata.json" :: notFountEndsWith
   }
 
   Given("""^there is no file at the url "(.*?)"$""") { (url: String) =>
-    // Implemented using stub_request in ruby implementation
-    // Need to figure out purpose of this stub request, may be called during validation later?
-  }
-
-//  Given("I have a metadata file called {String}") { filename: String =>
-//    // define
-//  }
-//
-//  And("it is stored at the url {String}") { url:String =>
-//    // define
-//  }
-//
-//  And("the metadata is stored at the url {String}") { url: String =>
-//    // define
-//  }
-//
-//  And("I have a file called {String} at the url {String}") { (fileName: String, url:String) =>
-//    // define
-//  }
-//
-//  And("there is no file at the url {String}") { url: String =>
-//    // define
-//  }
-//
-//  And("there is no file at the url {String}") { url: String =>
-//    // define
-//  }
-//
-//  When("I carry out CSVW validation") { () =>
-//    // define
-//  }
-//
-//  Then("there should not be errors") { () =>
-//    // define
-//  }
-//
-//  And("there should not be warnings") { () =>
-//    // define
-//  }
+    // notFoundStartsWith is a list which holds all the url startwith strings for which a 404 should be returned.
+    // This list is later used in setTestingBackend function to mock http calls
+    notFoundStartsWith = url :: notFoundStartsWith
+  }
+
+  Given("""^I have a metadata file called "([^"]*)"$""") { filename: String =>
+    val filePath = fixturesPath + filename
+    val metadata = Source.fromFile(filePath).getLines.mkString
+  }
+
+  And("""^the (schema|metadata) is stored at the url "(.*?)"$""") { (schemaType:String, url:String) =>
+    schemaUrl = url
+  }
 
+  And("""^I have a file called "(.*?)" at the url "(.*?)"$""") { (fileName: String, url:String) =>
+    val filePath = fixturesPath + fileName
```

Comment:
Class level properties?

Reply:
filePath is only applicable for the specific step definition and I don't think these must be made class level property. Whereas I have defined some other properties (like csv, metadata inside the step definition) which I have changed to class properties. 
Example scenario steps
		Given I have a CSV file called "csvw/test001.csv"
		And it is stored at the url "https://w3c.github.io/csvw/tests/test001.csv"
		And there is no file at the url "https://w3c.github.io/.well-known/csvm"